circle.yml

```diff
@@ -0,0 +1,72 @@
+# These environment variables must be set in CircleCI UI
+#
+# DOCKERHUB_REPO - docker hub repo, format: <username>/<repo>
+# DOCKER_EMAIL   - login info for docker hub
+# DOCKER_USER
+# DOCKER_PASS
+#
+machine:
+  services:
+    - docker
+
+  node:
+    version: 0.12.10
+
+dependencies:
+
+  # make sure to keep the docker cache dir
+  cache_directories:
+    - "~/docker"
+
+  override:
+    - docker info
+
+    # build the container, use circleci's docker cache workaround
+    # only use 1 image per day to keep the cache size from getting
+    # too big and slowing down the build
+    - I="image-$(date +%j).tgz"; if [[ -e ~/docker/$I ]]; then echo "Loading $I"; gunzip -c ~/docker/$I | docker load; fi
+
+    # create a version.json
+    - printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s"}\n' "$CIRCLE_SHA1" "$CIRCLE_TAG" "$CIRCLE_PROJECT_USERNAME" "$CIRCLE_PROJECT_REPONAME" > version.json
+
+    # build static assets
+    - npm install
+    - npm run lint
+    - npm run build
+    - npm test
+
+    # build the actual deployment container
+    - docker build -t app:build .
+
+    # Clean up any old images and save the new one
+    - I="image-$(date +%j).tgz"; mkdir -p ~/docker; rm ~/docker/*; docker save app:build | gzip -c > ~/docker/$I; ls -l ~/docker
+
+test:
+  override:
+    - >
+        docker run -it --rm --net=host 
```

Comment:
FWIW, with the `--rm` here, I get permission errors on CircleCI:

    Failed to destroy btrfs snapshot: operation not permitted


Also probably don't need the `-it` because that gives you an interactive terminal, and that doesn't get used here